hypothesis-python/tests/numpy/test_from_dtype.py

```diff
@@ -283,3 +284,21 @@ def condition(n):
         find_any(strat, condition)
     else:
         assert_no_examples(strat, condition)
+
+
+@pytest.mark.parametrize("dtype", STANDARD_TYPES)
+def test_resolves_and_varies_numpy_type(dtype):
+    @given(x=from_type(dtype.type))
+    def test_is_resolved_and_varied(x):
+        assert isinstance(x, dtype.type)
+        # Filter out those equal to the default value, to check that we see
+        # non-default values as well
+        assume(x != type(x)())
```

Comment:
Our `find_any(from_type(dtype.type), lambda x: x != type(x)())` test helper means you don't need to define a function internally here, and it's a bit more efficient too thanks to early stopping.